test/utils/asserts.cpp

```diff
@@ -5,7 +5,7 @@
 #include "asserts.hpp"
 #include <ostream>
 
-namespace fizzy
+namespace fizzy::test
```

Comment:
This will break [ADL](https://en.cppreference.com/w/cpp/language/adl).

Reply:
Not sure I understand, a this notation was already used in some files.

Reply:
I mean putting `operator<<` for class `fizzy::execution_result` in some other namespace breaks ADL.

Reply:
But if the PR compiles then nothing broke or am I missing something?

Reply:
Did you try output of failed `EXPECT_THAT(..., Result(x))` ? It might be broken.

With this change any `cout << result;` would not work unles it's inside `fizzy::test` or has `using namespace fizzy::test`.

I woul leave it in `fizzy`

Reply:
> Did you try output of failed EXPECT_THAT(..., Result(x)) ? It might be broken.
>
>With this change any cout << result; would not work unless it's inside fizzy::test or has using namespace fizzy::test.

All tests are like that.

Reply:
The code outputting `expected_result` in case of failure is not in the tests, it's in google.test.
This is what happens after this change:
```
Value of: execute(module, 0, {3, 6, 0})
Expected: result 5
  Actual: 32-byte object <00-6E 47-91 FC-7F 00-01 80-8C 81-02 00-00 00-00 88-8C 81-02 00-00 00-00 88-8C 81-02 00-00 00-00> (of type fizzy::execution_result)
```

Reply:
I don't really get how is it better for this reason to be in `fizzy` and not `fizzy::test`.

Reply:
When compiler sees `cout << result` it will include namespace of `execution_result` in the search for `operator<<` definition (so it will searchi in `fizzy`, but not in `fizzy::test`)

Reply:
How about this version of the PR then. Removed moving the `<<` operator.